src/common/components/select-repository-list/index.js

```diff
@@ -32,8 +33,10 @@ export class SelectRepositoryListComponent extends Component {
   componentWillReceiveProps(nextProps) {
     const repositoriesStatus = nextProps.repositoriesStatus;
     const ids = Object.keys(repositoriesStatus);
+    const owner = nextProps.user.login;
     if (ids.length && ids.every((id) => repositoriesStatus[id].success)) {
-      this.props.router.push('/dashboard');
+      this.props.dispatch(fetchUserSnapsIfNeeded(owner));
+      this.props.router.push('/dashboard/');
```

Comment:
Routes define `/dashboard` without trailing slash, so it would be good to keep it the same here.